### DISCUSSION TITLE
GitLab integrations stopped working

### DISCUSSION THREAD

**Discussion by mbrodala** [renovatebot/renovate#8587] [Request Help]
**What Renovate type, platform and version are you using?**

Hosted app with GitLab.

**Describe the bug**

Renovate is not working anymore: it does not suggest any updates anymore, adding new repositories does not have any effect, added repositories are not shown in the Renovate dashboard anymore.

**Relevant debug logs**

None.

**Have you created a minimal reproduction repository?**

Yes: https://gitlab.com/mbrodala/renovate-demo

**Additional context**

Dashboard when adding the `renovate-demo` repository:

![image](https://user-images.githubusercontent.com/5037116/107193661-5c34da80-69ef-11eb-9a29-dd41558ba323.png)

After closing the modal for adding repositories and refreshing:

![image](https://user-images.githubusercontent.com/5037116/107193778-8090b700-69ef-11eb-8303-794d010963e9.png)

Empty response in the https://v1.renovateapi.com/gitlab/installations Ajax request:

![image](https://user-images.githubusercontent.com/5037116/107193853-969e7780-69ef-11eb-92cf-7878465dbaa6.png)

**Comment by rarkins** [comment]
Unfortunately we needed to disable the GitLab aspect of the app after determining that GitLab's security model is not adequate to run a "shared" app. We described what we found [here](https://docs.renovatebot.com/gitlab-bot-security/) and also created a simple template for self-hosted that you can hopefully migrate to.

**Reply by mbrodala** [reply]
Thanks for the info, I didn't notice this at all. Maybe a notice or similar should be shown for users of GitLab repositories in the dashboard?

In any case I'll look into the options we have now.

**Comment by mbrodala** [comment]
Just saw https://github.com/renovatebot/app-support/issues/106

**Reply by yangm97** [reply]
@rarkins now that the app-support repo got archived, is this discussion the "spiritual successor" for that issue?

**Reply by rarkins** [reply]
@yangm97 I guess so.

FYI we're still in the same problem where GitLab's security model for CI tokens is not compatible with the concept of shared bots.

**Reply by gperdomor** [reply]
Now with the new security of `CI_JOB_TOKEN` available on gitlab.com, the gitlab app can be reenabled?  https://about.gitlab.com/releases/gitlab-com/

**Reply by rarkins** [reply]
Maybe once they enable it by default

**Reply by gperdomor** [reply]
@rarkins 😢 seems to be no plans to enable that by default

**Reply by rarkins** [reply]
Until then, users would be vulnerable, no? We could *recommend* they flip the setting, but they may not.
